### PR TITLE
chore: add schema docs to AWS::Serverless::StateMachine

### PR DESCRIPTION
**Pull request by aaythapa**
### Issue #, if available

#1133 

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [ ] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

### Examples?

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/schema/aws_serverless_statemachine.py [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
+    Description: Optional[PassThrough] = scheduleeventproperties("Description")
+    Enabled: Optional[bool] = scheduleeventproperties("Enabled")
+    Input: Optional[PassThrough] = scheduleeventproperties("Input")
+    Name: Optional[PassThrough] = scheduleeventproperties("Name")
+    RetryPolicy: Optional[PassThrough] = scheduleeventproperties("RetryPolicy")
+    Schedule: Optional[PassThrough] = scheduleeventproperties("Schedule")
+    State: Optional[PassThrough] = scheduleeventproperties("State")
 
 
 class ScheduleEvent(BaseModel):
```
Comment (hoffa) [inline samtranslator/schema/aws_serverless_statemachine.py:38]: Also all `Type`/`Properties` for events.

File: samtranslator/schema/aws_serverless_statemachine.py [hunk 2]
```diff
@@ -77,12 +90,12 @@ class CloudWatchEvent(BaseModel):
...
-    EventBusName: Optional[PassThrough]
-    Input: Optional[PassThrough]
-    InputPath: Optional[PassThrough]
-    Pattern: Optional[PassThrough]
-    RetryPolicy: Optional[PassThrough]
+    DeadLetterConfig: Optional[DeadLetterConfig] = eventbridgeruleeventproperties("DeadLetterConfig")
+    EventBusName: Optional[PassThrough] = eventbridgeruleeventproperties("EventBusName")
+    Input: Optional[PassThrough] = eventbridgeruleeventproperties("Input")
+    InputPath: Optional[PassThrough] = eventbridgeruleeventproperties("InputPath")
+    Pattern: PassThrough = eventbridgeruleeventproperties("Pattern")
```
Comment (hoffa) [inline samtranslator/schema/aws_serverless_statemachine.py:97]: Why was this changed from `Optional[PassThrough]` to `PassThrough`?

Comment (hoffa) [reply samtranslator/schema/aws_serverless_statemachine.py:97]: Oh it's supposed to be required: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-statemachine-statemachineeventbridgerule.html#sam-statemachine-statemachineeventbridgerule-pattern

Great catch! 👍

**Review by hoffa — APPROVED** [review]

**Review by GavinZZ — APPROVED** [review]
